File: AntiGravity/lib/models/company.dart
enum SubscriptionTier { standard, professional, enterprise, custom }

class Company {
  final String name;
  final String code; // e.g. "DINE"
  final String email;
  final String mobile;
  final SubscriptionTier subscriptionTier;
  final List<String> activeModuleIds; // Gated modules
  final DateTime createdAt;
  final String? primaryColorHex;
  final String? secondaryColorHex;

  Company({
    required this.name,
    required this.code,
    required this.email,
    required this.mobile,
    this.subscriptionTier = SubscriptionTier.standard,
    required this.activeModuleIds,
    DateTime? createdAt,
    this.primaryColorHex,
    this.secondaryColorHex,
  }) : createdAt = createdAt ?? DateTime.now();

  String get tierName {
    switch (subscriptionTier) {
      case SubscriptionTier.standard:
        return 'Standard';
      case SubscriptionTier.professional:
        return 'Professional';
      case SubscriptionTier.enterprise:
        return 'Enterprise';
      case SubscriptionTier.custom:
        return 'Custom / Enterprise+';
    }
  }

  // Generate default modules for a subscription tier
  static List<String> getDefaultModulesForTier(SubscriptionTier tier) {
    switch (tier) {
      case SubscriptionTier.standard:
        return ['admin', 'communication', 'sales', 'crm'];
      case SubscriptionTier.professional:
        return ['admin', 'communication', 'sales', 'crm', 'purchase', 'inventory', 'finance', 'hrm', 'project'];
      case SubscriptionTier.enterprise:
      case SubscriptionTier.custom:
        return [
          'admin', 'crm', 'sales', 'purchase', 'inventory', 'manufacturing', 
          'finance', 'hrm', 'project', 'scm', 'qms', 'maintenance', 'pos', 
          'ecommerce', 'analytics', 'communication', 'ai', 'security', 'mobile', 'industry'
        ];
    }
  }

  Company copyWith({
    String? name,
    String? email,
    String? mobile,
    SubscriptionTier? subscriptionTier,
    List<String>? activeModuleIds,
    String? primaryColorHex,
    String? secondaryColorHex,
  }) {
    return Company(
      name: name ?? this.name,
      code: code,
      email: email ?? this.email,
      mobile: mobile ?? this.mobile,
      subscriptionTier: subscriptionTier ?? this.subscriptionTier,
      activeModuleIds: activeModuleIds ?? this.activeModuleIds,
      createdAt: createdAt,
      primaryColorHex: primaryColorHex ?? this.primaryColorHex,
      secondaryColorHex: secondaryColorHex ?? this.secondaryColorHex,
    );
  }
}

File: ERP Claude K/backend/models/Company.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Company = sequelize.define('Company', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },
  company_code: {
    type: DataTypes.STRING(20),
    allowNull: false,
    unique: true,
    validate: {
      len: [3, 20],
      isAlphanumeric: true
    }
  },
  company_name: {
    type: DataTypes.STRING(255),
    allowNull: false
  },
  legal_name: {
    type: DataTypes.STRING(255)
  },
  email: {
    type: DataTypes.STRING(255),
    allowNull: false,
    validate: {
      isEmail: true
    }
  },
  phone: {
    type: DataTypes.STRING(20)
  },
  website: {
    type: DataTypes.STRING(255)
  },
  logo_url: {
    type: DataTypes.STRING(500)
  },
  
  // Address
  address_line1: {
    type: DataTypes.STRING(255)
  },
  address_line2: {
    type: DataTypes.STRING(255)
  },
  city: {
    type: DataTypes.STRING(100)
  },
  state: {
    type: DataTypes.STRING(100)
  },
  country: {
    type: DataTypes.STRING(100)
  },
  postal_code: {
    type: DataTypes.STRING(20)
  },
  
  // Tax & Legal
  gst_number: {
    type: DataTypes.STRING(50)
  },
  tax_number: {
    type: DataTypes.STRING(50)
  },
  pan_number: {
    type: DataTypes.STRING(50)
  },
  
  // Subscription
  subscription_plan: {
    type: DataTypes.ENUM('trial', 'basic', 'standard', 'premium', 'enterprise'),
    defaultValue: 'trial'
  },
  subscription_start: {
    type: DataTypes.DATE
  },
  subscription_end: {
    type: DataTypes.DATE
  },
  max_users: {
    type: DataTypes.INTEGER,
    defaultValue: 5
  },
  max_storage_gb: {
    type: DataTypes.INTEGER,
    defaultValue: 10
  },
  
  // Features (JSON field for enabled features)
  enabled_features: {
    type: DataTypes.JSONB,
    defaultValue: {
      crm: true,
      sales: true,
      purchase: false,
      inventory: false,
      manufacturing: false,
      finance: false,
      hrm: false,
      projects: false,
      scm: false,
      qms: false,
      maintenance: false,
      pos: false,
      ecommerce: false,
      analytics: true,
      chat: true
    }
  },
  
  // Settings
  settings: {
    type: DataTypes.JSONB,
    defaultValue: {
      currency: 'INR',
      timezone: 'Asia/Kolkata',
      date_format: 'DD/MM/YYYY',
      language: 'en',
      fiscal_year_start: '04-01'
    }
  },
  
  // Status
  is_active: {
    type: DataTypes.BOOLEAN,
    defaultValue: true
  },
  is_verified: {
    type: DataTypes.BOOLEAN,
    defaultValue: false
  },
  
  // Metadata
  onboarding_completed: {
    type: DataTypes.BOOLEAN,
    defaultValue: false
  },
  notes: {
    type: DataTypes.TEXT
  }
}, {
  tableName: 'companies',
  indexes: [
    { fields: ['company_code'], unique: true },
    { fields: ['email'] },
    { fields: ['subscription_plan'] },
    { fields: ['is_active'] }
  ]
});

module.exports = Company;
